enhancements/proposals/2549-scylladbcluster-backup-repair/README.md

```diff
@@ -0,0 +1,457 @@
+
+#### Auth tokens
+
+ScyllaDBCluster controller, running as part of the operator, will generate the auth token secret and configure the created ScyllaDBDatacenters with an internal annotation `internal.scylla-operator.scylladb.com/scylladb-manager-agent-auth-token-override-secret-ref`.
+ScyllaDBDatacenter controller will first try to get the auth token from the custom config in an unchanged manner. If the custom config is absent or does not specify the auth token, it will try to get the auth token from the Secret specified in the annotation. 
+In case the annotation is absent, it will retain the already generated token or generate it in an unchanged manner.
+
+#### Finalizers
+
+Both the ScyllaDBCluster controller and the ScyllaDBDatacenter controller, running as part of the manager-controller, will set a `scylla-operator.scylladb.com/scylladb-manager-cluster` finalizer on the reconciled object so that the clusters are deleted from ScyllaDB Manager state on deletion [#1930][2].
+
+#### Status conditions
+
+Both the ScyllaDBCluster controller and the ScyllaDBDatacenter controller, running as part of the manager-controller, will wait for the reconciled object to be rolled out to register it with ScyllaDB Manager.
+
+### Scheduling of backup and repair tasks with ScyllaDB Manager
+
+We introduce a new API, `ScyllaDBManagerTask.scylla.scylladb.com/v1alpha1`, which allows users to schedule backup and repair tasks for ScyllaDBClusters and ScyllaDBDatacenters registered with ScyllaDB Manager.
+The existing manager-controller application will be extended with a dedicated controller reconciling ScyllaDBManagerTask objects, responsible for scheduling the specified tasks with ScyllaDB Manager.
+
+Backup and repair tasks defined in `scyllaclusters.scylla.scylladb.com/v1.spec.backups` and `scyllaclusters.scylla.scylladb.com/v1.spec.repairs` will no longer be reconciled directly by manager-controller.
+Instead, the existing ScyllaCluster translation controller, running as part of the operator, is extended to convert the backup and repair tasks defined in `scyllaclusters.scylla.scylladb.com/v1.spec.backups` and `scyllaclusters.scylla.scylladb.com/v1.spec.repairs` into the ScyllaDBManagerTask objects and apply them.
+
+ScyllaDBManagerTask objects, created by the translation controller, adhere to the following naming convention to avoid name collisions:
+- Tasks created from `scyllaclusters.scylla.scylladb.com/v1.spec.backups`: SHA256(`scyllaclusters.scylla.scylladb.com/v1.name`, "backup", `scyllaclusters.scylla.scylladb.com/v1.spec.backups[].name`).
+- Tasks created from `scyllaclusters.scylla.scylladb.com/v1.spec.repairs`: SHA256(`<scyllaclusters.scylla.scylladb.com/v1.name>`, "repair", `scyllaclusters.scylla.scylladb.com/v1.spec.repairs[].name`).
+SHA256 hash is to reduce the length of the produced name, as both ScyllaCluster's and corresponding task's name could have max allowed name length. Hexadecimal representation of SHA256 is 64 characters.
```

Comment:
Is the only reason why we wouldn't just apply a `backup-` or `restore-` prefix to a preexisting name that we might be exceeding the max length?

I'd be inclined to keep the generated names readable. And if collisions because of truncation are a concern, we can solve that by adding an alphanum stable suffix similar t ohttps://github.com/scylladb/scylla-operator/pull/2505.

Reply:
> Is the only reason why we wouldn't just apply a backup- or restore- prefix to a preexisting name that we might be exceeding the max length?

Yes.

I suppose something like below is also an option, with the risk of exceeding max length, which is a bit silly of a reason to error on reconciliation imo. It's an under-the-hood object when created in ScyllaCluster translation, so is readability really a concern?

```
		nameSuffix, err := naming.GenerateNameHash(string(scyllav1alpha1.ScyllaDBManagerTaskTypeBackup), backupSpec.Name)
		if err != nil {
			return progressingConditions, fmt.Errorf("can't generate ScyllaDBManagerTask name suffix: %w", err)
		}
		name := fmt.Sprintf("%s-%s", sc.Name, nameSuffix)
```

Reply:
done